### PR TITLE
local ports: use cache lock (#9342)

### PR DESCRIPTION
**Pull request by Beuc**
Cf. #9342

### CODE REVIEW

File: tools/system_libs.py [hunk 1]
```diff
@@ -1437,23 +1437,27 @@ def fetch_project(name, url, subdir, is_tarbz2=False, sha512hash=None):
...
+            port = ports.ports_by_name[name]
+            if not hasattr(port, 'SUBDIR'):
+              logger.error('port %s lacks .SUBDIR attribute, which we need in order to override it locally, please update it' % name)
+              sys.exit(1)
+            subdir = port.SUBDIR
+            logger.warning('grabbing local port: ' + name + ' from ' + path + ' to ' + fullname + ' (subdir: ' + subdir + ')')
+            shared.try_delete(fullname)
+            shutil.copytree(path, os.path.join(fullname, subdir))
+            Ports.clear_project_build(name)
+            return
```
Comment (sbc100) [inline tools/system_libs.py]: Can you leave the return statement outside (after) the try/finally.. it makes the early return pattern more clear.

Also, the the first two lines of the try block and stay outside too I think.

File: tools/system_libs.py [hunk 2]
```diff
@@ -1437,23 +1437,27 @@ def fetch_project(name, url, subdir, is_tarbz2=False, sha512hash=None):
...
+              shared.exit_with_error('%s is not a known port' % name)
+            port = ports.ports_by_name[name]
+            if not hasattr(port, 'SUBDIR'):
+              logger.error('port %s lacks .SUBDIR attribute, which we need in order to override it locally, please update it' % name)
+              sys.exit(1)
+            subdir = port.SUBDIR
+            logger.warning('grabbing local port: ' + name + ' from ' + path + ' to ' + fullname + ' (subdir: ' + subdir + ')')
+            shared.try_delete(fullname)
+            shutil.copytree(path, os.path.join(fullname, subdir))
+            Ports.clear_project_build(name)
```
Comment (sbc100) [inline tools/system_libs.py]: Presumably its these three lines that require the lock?

Comment (Beuc) [reply tools/system_libs.py]: The SDL2 port deletes depending ports prior (re)building, so I'm afraid that locking only those lines will lead to build failures like SDL2 (process1) > SDL2_image (p1) > SDL2 (process2) > SDL2_ttf (p1) > missing SDL2.h.
That's basically why I didn't try to be specific about the lock.
I cannot confirm right now due to #9402.

File: tools/system_libs.py (hunk 1 above)
Comment (Beuc) [reply tools/system_libs.py]: Done :)

File: tools/system_libs.py (hunk 2 above)
Comment (kripken) [reply tools/system_libs.py]: This seems reasonable to me. Basically, any local-ports work should be locked.

**Comment by kripken** [comment]
Something weird happened with the tests here, but this should be safe to merge...

**Pull request merged**